### ISSUE TITLE
Feature request: "Paste file here" 

### ISSUE THREAD

**Issue by andyjda** [Alexander-Miller/treemacs#994]
I have created [a pull-request](https://github.com/Alexander-Miller/treemacs/pull/995) with a possible implementation.

Rationale: in a typical GUI-interface, one can copy and move files around with just two mouse-events or key-commands inside the appropriate directories.
I might be missing something, but it seems to me that treemacs has an additional step: if I want to move a file from one directory to another directory within the same workspace, I'll have to:

1. copy or move the file (`treemacs--copy-or-move`). This generates the prompt "Copy to: " or "Move to: " in the minibuffer
2. navigate (via scrolling or keybindings) to the new location within treemacs
3. `treemacs-copy-absolute-path-at-point`
4. paste the copied-path into the minibuffer, so that `treemacs--copy-or-move` can finish its job

My idea is to unite steps 3 and 4 into one "Paste here" function, which will make this workflow easier: copy/move from one location, navigate to the next location, paste.

[See the pull-request](https://github.com/Alexander-Miller/treemacs/pull/995) for the implementation. It's really simple and could probably use some polishing, but it's a good overview of the intended use. Some notes:
 - right now the function `treemacs--paste-point-to-minibuffer` assumes that we are running `treemacs--copy-or-move`, and simply pastes the path at point to the minibuffer. The "paste here" functionality could have a more complex implementation: for example, check if the last kill is a file path, and if it is, copy that file here. But I don't know if there's much value in that, and I can see the value of keeping the implementation simple
 - the check to make sure that we are running `treemacs--copy-or-move` is just to check the minibuffer prompt. If the prompt is "Copy to: " or "Move to: ", the "Paste here" menu item is displayed.
 - I added a possible keybinding to the functionality. I picked "y v" as a nod to the CUA binding C-v for pasting, but I'm open to other ideas.

**Comment by Alexander-Miller** [comment]
Something's off about your workflow description. Nothing after the first step is supposed to happen as you describe it. You should not be navigating in treemacs when the prompt is seen, and certainly not to grab an entire absolute path.

What should happen is that focus switches to the minibuffer and  you enter the desired location right in there. Even basic Emacs without any additional packages will offer you tab-completion to select files, using the likes of helm or vertico will make it even easier. In addition default-directoy is set to the nearest path at point so it's not like you even need to type out the full path starting from `~/`.

Mouse events are also supported - you can move files by dragging them around. It's not very discoverable because there's no visual feedback like in other applications, but that is simply an Emacs limitation.

I'd be more interested in finding out why moving files is so much more tedious for you than implementing such a strange workaround.

**Comment by andyjda** [comment]
These are good points.
Let me describe the use-case that made me think about this solution, and why I found the minibuffer / dragging approaches sub-optimal for it.

I had to copy some files from one project to another project. Both projects were nested deep within sub-directories, and their "most-recent" common parent-directory was the home directory:

- `~/dir_A/subdir_1/subdir_2/subdir_3/project_a/subdir_1/subdir_2/file_to_copy`
- `~/dir_B/subdir_1/subdir_2/subdir_3/project_B/subdir_1/subdir_2/dir_to_paste_file_in`

I also wasn't very familiar with the directory structure of the projects yet. I found that, since I had both projects within Treemacs, simply copying the path of the new location was easier than writing it down in the minibuffer. Even with tab completion (and even if I knew the directory structure perfectly), writing the new path would take significantly more steps than copy-pasting it.
Having default-directory set to the nearest path at point is a very good default, in my opinion, but in cases like this (moving/copying files across projects), it's not much help.
Even in cases where files are being moved/copied within the same project, my new "Paste here" solution takes about the same number of events as the best-case interaction with minibuffer:
 - "Copy file" -> switch to minibuffer interface and press first letter of the new directory -> tab complete -> Enter 
 - "Copy file" -> within the treemacs interface, navigate to new directory -> "Paste here" -> Enter
 
The "Paste here" solution has the added benefit of keeping the whole interaction within the same treemacs interface, and being more consistently visually-based (though I do think the minibuffer approach is valuable: both options should be available).


About dragging files:
The most important limitation is that, as far as I understand, dragging only allows users to _move_ files, not _copy_ them.
There's other issues that might be due to my set-up: I use a trackpad, which makes it a bit harder to hold down and scroll at the same time. It's easier to make mistakes: if I just stop pressing down, my file is gone into the wrong directory. Also, if the new destination for the file is not visible, I can not scroll into it with the trackpad while pressing down: in most applications, usually dragging towards the top/bottom of the visible space automatically scrolls the visible space. I think that would be overkill to implement, so my solution is to scroll using keybindings, but I find  that to be not as convenient as the simpler "Paste here" approach.
